src/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -869,11 +878,27 @@ else if (itemChildName.equals("value")) {
         return tags;
     }
 
-    protected QuestionDef parseControl(IFormElement parent, Element e, int controlType) {
-        return parseControl (parent, e, controlType, null);
+    private QuestionDef parseControl(IFormElement parent, Element e, int controlType) {
+        return parseControl (parent, e, controlType, null, null);
+    }
+
+    private QuestionDef parseControl(IFormElement parent, Element e, int controlType, List<String> additionalUsedAtts) {
+        return parseControl (parent, e, controlType, additionalUsedAtts, null);
     }
 
-    protected QuestionDef parseControl(IFormElement parent, Element e, int controlType, List<String> additionalUsedAtts) {
+    /**
+     * Parses a form control element into a {@link org.javarosa.core.model.QuestionDef} and attaches it to its parent.
+     *
+     * @param parent                the form control element's parent
+     * @param e                     the form control element to parse
+     * @param controlType           one of the control types defined in {@link org.javarosa.core.model.Constants}
+     * @param additionalUsedAtts    attributes specific to the control type
+     * @param passedThroughAtts     attributes specific to the control type that should be passed through to
+     *                              additionalAttributes for historical reasons
+     * @return                      a {@link org.javarosa.core.model.QuestionDef} representing the form control element
+     */
+    private QuestionDef parseControl(IFormElement parent, Element e, int controlType, List<String> additionalUsedAtts,
```

Comment:
This doc comment is quite a nice improvement.

Reply:
🎉